### ISSUE TITLE
False positive for whitespace/indent_namespace with member initializer lists

### ISSUE THREAD

**Issue by marinedayo** [cpplint/cpplint#372]
I have encountered a few patterns of false positives for `whitespace/indent_namespace`.
All of them occur with the member initializer lists, which have multiple lines.

I hope this will be helpful for improvement.

cpplint version: 2.0.2

```
$ cpplint --version
Cpplint fork (https://github.com/cpplint/cpplint)
cpplint 2.0.2
Python 3.12.9 (main, Feb  4 2025, 14:38:38) [Clang 16.0.0 (clang-1600.0.26.6)]
```

case1: constructor with an empty body

```c++
namespace cpplint_issue {

class TestIndentNamespaceWithEmptyBody {
 public:
  explicit TestIndentNamespaceWithEmptyBody(const int a);

 private:
  const int number;
  const int looooooong_attr1;
  const int looooooong_attr2;
  const int looooooong_attr3;
  const int looooooong_attr4;
};

TestIndentNamespaceWithEmptyBody::TestIndentNamespaceWithEmptyBody(const int a)
    : number(a),
      looooooong_attr1(0),
      looooooong_attr2(0),
      looooooong_attr3(0),
      looooooong_attr4(0) {}  // <- This line causes a false positive.

}  // namespace cpplint_issue
```

case2: use inline specifier

```c++
namespace cpplint_issue {

class TestIndentNamespaceWithInline {
 public:
  explicit TestIndentNamespaceWithInline(const int a);

 private:
  const int number;
  const int looooooong_attr1;
  const int looooooong_attr2;
  const int looooooong_attr3;
  const int looooooong_attr4;
};

inline TestIndentNamespaceWithInline::TestIndentNamespaceWithInline(const int a)
    : number(a),             // <- These lines cause a false positive.
      looooooong_attr1(0),   // <-
      looooooong_attr2(0),   // <-
      looooooong_attr3(0),   // <-
      looooooong_attr4(0) {  // <-
  // do nothing
}

}  // namespace cpplint_issue
```

case3: with a struct variable

```c++
namespace cpplint_issue {

struct Params {
  int number;
};

class TestIndentNamespaceWithStructMember {
 public:
  explicit TestIndentNamespaceWithStructMember(const int a);

 private:
  const Params params;
  const int looooooong_attr1;
  const int looooooong_attr2;
  const int looooooong_attr3;
  const int looooooong_attr4;
};

TestIndentNamespaceWithStructMember::TestIndentNamespaceWithStructMember(const int a)
    : params({a}),
      looooooong_attr1(0),   // <- These lines cause a false positive.
      looooooong_attr2(0),   // <-
      looooooong_attr3(0),   // <-
      looooooong_attr4(0) {  // <-
  // do nothing
}

}  // namespace cpplint_issue
```

**Comment by wushenrong** [comment]
This also happens with braces when using member initializer lists:

```cpp
namespace cpplint_issue {

class TestIndentNamespaceWithEmptyBody {
 public:
  explicit TestIndentNamespaceWithEmptyBody(const int a);

 private:
  const int number;
  const int looooooong_attr1;
  const int looooooong_attr2;
  const int looooooong_attr3;
  const int looooooong_attr4;
};

TestIndentNamespaceWithEmptyBody::TestIndentNamespaceWithEmptyBody(const int a)
    : number{a},              // <- These lines causes a false positive.
      looooooong_attr1{0},    // <-
      looooooong_attr2{0},    // <-
      looooooong_attr3{0},    // <-
      looooooong_attr4{0} {}  // <-

}  // namespace cpplint_issue
```

**Comment by aaronliu0130** [comment]
I've fixed most cases mentioned here locally, but I'm struggling to figure out how to fix the {initializer list/uniform initialization} case. Perhaps I have the contents of the member initializer list's braces consumed.

**Comment by wushenrong** [comment]
Adding putting more initializer list cases that causes the false postive

Member initializer list with braces on the same line. Using parenthesizes does not cause a false positive. (It's implied in my last comment but putting it here for completion.)
```cpp
namespace example_namespace {

class ExampleClass {
  ExampleClass(int a, int b);

 private:
  int number_one;
  int number_two;
};

ExampleClass::ExampleClass(int a, int b)
    : number_one{a}, number_two{b} {} // <- This line causes a false positive.

}  // namespace example_namespace
```

Mixing braces and parenthesizes in member initializer list. This triggers if the member names are different lines except for the first member in the list. It does not matter which member uses braces or parenthesizes.
```cpp
namespace example_namespace {

class ExampleClass {
  ExampleClass(int a, int b);

 private:
  int number_one;
  int number_two;
};

ExampleClass::ExampleClass(int a, int b)
    : number_one(a), number_two{b} {} // <- This line causes a false positive.

}  // namespace example_namespace
```

**Comment by aaronliu0130** [comment]
Thanks. I've identified the issue for that case and will reopen this issue if the PR gets merged before I make a fix for that case. (And whether braces are mixed doesn't matter; all that matters is the closing brace for the body is on the same line as an initializer list item that uses outer uniform initialization.)

**Comment by aaronliu0130** [comment]
@wushenrong Should be fixed, try again!

**Comment by wushenrong** [comment]
Sorry about the long wait, I can confirm the false positive does not get triggered with the latest development version of cpplint.

Edit: One more false positive appears when a base constructor is called in the member initializer list, didn't add this because it probably would be added in another pr. Member variables are fine.

```cpp
namespace example_namespace {

DerivedClass::DerivedClass(int base_member, int derived_member)
    // The line below triggers a false positive, this includes mixing braces
    : BaseClass{base_member}, derived_member_{derived_member} {}
    // The line below does not
    : BaseClass(base_member_variable), derived_member_(derived_member) {}

}  // namespace example_namespace

```
